src/main/java/net/sf/jabref/model/database/BibDatabaseMode.java

```diff
@@ -2,34 +2,59 @@
 
 import java.util.Locale;
 
+/**
+ *
+ * An enum which contains the possible {@link BibDatabase} Modes.
+ * Possible are BibTeX and BibLaTeX.
+ *
+ */
 public enum BibDatabaseMode {
     BIBTEX,
     BIBLATEX;
 
+    /**
+     *
+     * @return the name of the current mode as String
+     */
     public String getFormattedName() {
-        if(this == BIBTEX) {
-            return "BibTeX";
-        } else {
-            return "BibLaTeX";
-        }
+        return (this == BIBTEX ? "BibTeX" : "BibLateX");
     }
 
+    /**
+     * Returns the opposite mode of the current mode as {@link BibDatabaseMode}.
+     *
+     * @return BibLaTeX if the current mode is BIBTEX, BibTeX else
+     */
     public BibDatabaseMode getOppositeMode() {
-        if(this == BIBTEX) {
-            return BIBLATEX;
-        } else {
-            return BIBTEX;
-        }
+        return (this == BIBTEX ? BIBLATEX : BIBTEX);
     }
 
+    /**
+     * Returns the mode from a given boolean.
+     *
+     * @param isBibLatex
+     * @return BIBLATEX if isBibLatex is true, else BIBTEX
+     */
     public static BibDatabaseMode fromPreference(boolean isBibLatex) {
         return isBibLatex ? BIBLATEX : BIBTEX;
     }
 
+    /**
+     * Returns the enum type that equals the given string.
+     * The use of capital and small letters in the string doesn't matter.
+     *
+     * @param   data
+     * @return  BIBTEX, if the string is bibtex
```

Comment:
What happens if the string neither "bibtex" nor "biblatex". You can write in the description that this will invoke `{@link valueOf(Strring value)}`.